dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -142,6 +146,94 @@ static ClassDumper create(List<Path> jarFilePaths) {
     return methodReferences.build();
   }
 
+  static SymbolsInFile scanClassSymbolTable(JavaClass javaClass) {
+    SymbolsInFile.Builder symbolTableBuilder = SymbolsInFile.builder();
+    ImmutableSet.Builder<MethodSymbolReference> methodReferences =
+        symbolTableBuilder.methodReferencesBuilder();
+    ImmutableSet.Builder<FieldSymbolReference> fieldReferences =
+        symbolTableBuilder.fieldReferencesBuilder();
+
+    // TODO(suztomo): Read class references (inheritance) from javaClass file
+    ImmutableSet.Builder<ClassSymbolReference> classReferences =
+        symbolTableBuilder.classReferencesBuilder();
+
+    String sourceClassName = javaClass.getClassName();
+    ConstantPool constantPool = javaClass.getConstantPool();
+    Constant[] constants = constantPool.getConstantPool();
+    for (Constant constant : constants) {
+      if (constant == null) {
+        continue;
+      }
+      byte constantTag = constant.getTag();
+      switch (constantTag) {
+        case Const.CONSTANT_Methodref:
+          ConstantMethodref constantMethodref = (ConstantMethodref) constant;
+          methodReferences.add(constantToMethodReference(constantMethodref, constantPool,
+              sourceClassName));
+          break;
+        case Const.CONSTANT_Fieldref:
+          ConstantFieldref constantFieldref = (ConstantFieldref) constant;
+          fieldReferences.add(constantToFieldReference(constantFieldref, constantPool,
+              sourceClassName));
+          break;
+        case Const.CONSTANT_Class:
+          // TODO(suztomo): handle class reference
+          break;
+        default:
+          break;
+      }
+    }
+
+    return symbolTableBuilder.build();
+  }
+
+  static ConstantNameAndType constantNameAndTypeFromConstantCP(
+      ConstantCP constantCP, ConstantPool constantPool) {
+    int nameAndTypeIndex = constantCP.getNameAndTypeIndex();
+    Constant constantAtNameAndTypeIndex = constantPool.getConstant(nameAndTypeIndex);
+    if (!(constantAtNameAndTypeIndex instanceof ConstantNameAndType)) {
+      // This constant_pool entry must be a CONSTANT_NameAndType_info
+      // as specified https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.4.2
+      throw new RuntimeException(
```

Comment:
Per https://engdoc.corp.google.com/eng/doc/devguide/java/practices/exceptions.md?cl=head "Treat Throwable, Exception, Error, and RuntimeException as if they were abstract: avoid constructing or throwing a new instance of any of these types directly."

This might be better as a checked exception anyway. I'd need to think about that.